File: sample-app-java+push-notification/src/main/java/com/cometchat/sampleapp/java/fcm/ui/fragments/CallDetailsTabParticipantsFragment.java
package com.cometchat.sampleapp.java.fcm.ui.fragments;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.cometchat.calls.model.CallLog;
import com.cometchat.calls.model.CallUser;
import com.cometchat.sampleapp.java.fcm.databinding.FragmentCallDetailsTabBinding;
import com.cometchat.sampleapp.java.fcm.ui.adapters.CallDetailsTabParticipantsAdapter;
import com.google.gson.Gson;

public class CallDetailsTabParticipantsFragment extends Fragment {

    private FragmentCallDetailsTabBinding binding;
    private CallLog callLog;

    public CallDetailsTabParticipantsFragment() {
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (getArguments() != null) {
            callLog = new Gson().fromJson(getArguments().getString("callLog"), CallLog.class);
            callLog.setInitiator(new Gson().fromJson(getArguments().getString("initiator"), CallUser.class));
            callLog.setReceiver(new Gson().fromJson(getArguments().getString("receiver"), CallUser.class));
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentCallDetailsTabBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding.rvCallDetails.setLayoutManager(new LinearLayoutManager(getContext()));
        binding.rvCallDetails.setAdapter(new CallDetailsTabParticipantsAdapter(callLog));
    }
}

File: sample-app-kotlin+push-notification/src/main/java/com/cometchat/sampleapp/kotlin/fcm/ui/fragments/CallDetailsTabParticipantsFragment.kt
package com.cometchat.sampleapp.kotlin.fcm.ui.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.cometchat.calls.model.CallLog
import com.cometchat.calls.model.CallUser
import com.cometchat.sampleapp.kotlin.fcm.databinding.FragmentCallDetailsTabBinding
import com.cometchat.sampleapp.kotlin.fcm.ui.adapters.CallDetailsTabParticipantsAdapter
import com.google.gson.Gson

class CallDetailsTabParticipantsFragment : Fragment() {

    private lateinit var binding: FragmentCallDetailsTabBinding
    private lateinit var callLog: CallLog

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (arguments != null) {
            callLog = Gson().fromJson(
                requireArguments().getString("callLog"), CallLog::class.java
            )
            callLog.initiator = Gson().fromJson(
                requireArguments().getString("initiator"), CallUser::class.java
            )
            callLog.receiver = Gson().fromJson(
                requireArguments().getString("receiver"), CallUser::class.java
            )
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        binding = FragmentCallDetailsTabBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.rvCallDetails.layoutManager = LinearLayoutManager(
            context
        )
        binding.rvCallDetails.adapter = CallDetailsTabParticipantsAdapter(
            callLog
        )
    }
}
